File: src/geometries/Triangle.ts
import { Geometry } from './Geometry';

export class Triangle extends Geometry {
  constructor() {
    const vertices = new Float32Array([
      0.0, 1.0, 0.0,
     -1.0, -1.0, 0.0,
      1.0, -1.0, 0.0
    ]);
    super({ vertices });
  }
}

File: iEngine-Native/Engine/src/geometries/Triangle.cpp
#include "iengine/geometries/Triangle.h"

namespace iengine {
    Triangle::Triangle() : Geometry(
        // 顶点数据 (x, y, z) - 与Web版本保持一致
        {
             0.0f,  1.0f, 0.0f,  // 顶点
            -1.0f, -1.0f, 0.0f,  // 左下
             1.0f, -1.0f, 0.0f   // 右下
        }
    ) {}
}